src-common/main/scala/uk/gov/hmrc/play/audit/http/HttpAuditing.scala

```diff
@@ -58,41 +61,45 @@ trait HttpAuditing {
     override def apply(
       verb     : String,
       url      : URL,
-      headers  : Seq[(String, String)],
-      body     : Option[HookData],
-      responseF: Future[HttpResponse]
+      request  : RequestData,
+      responseF: Future[ResponseData]
     )(implicit
       hc: HeaderCarrier,
-      ec : ExecutionContext
-    ): Unit = {
-      val request = HttpRequest(verb, url.toString, headers, body, now())
-      responseF.map {
-        response => audit(request, response)
-      }.recover {
-        case e: Throwable => auditRequestWithException(request, e.getMessage)
-      }
-    }
+      ec: ExecutionContext
+    ): Unit =
+      // short-circuit the payload creation (and truncation log warning)
+      if (auditConnector.isEnabled) {
+        val httpRequest = HttpRequest(verb, url.toString, request.headers, request.body, now())
+        responseF.map {
+          response => audit(httpRequest, response)
+        }.recover {
+          case e: Throwable => auditRequestWithException(httpRequest, e.getMessage)
+        }
+      } else
+        // TODO is this needed? (Would have been logged if we hadn't short-circuited)
```

Comment:
This questions applies to the log inside [AuditConnector](https://github.com/hmrc/play-auditing/blob/master/src-common/main/scala/uk/gov/hmrc/play/audit/http/connector/AuditConnector.scala#L102) too. We get a log entry for every call when disabled. Would a single log to warn it has been disabled (without requestId and sessionId) be sufficient?